framework/web/User.php

```diff
@@ -234,6 +234,47 @@ public function setIdentity($identity)
         $this->_access = [];
     }
 
+    /**
+     * Sets new [[authKey]] for existing user.
+     * 
+     * If user is not currently logged in or new identity doesn't match currently logged in user,
+     * this method does not login new user nor updates [[authKey]]. You should usually use 
+     * [[switchIdentity()]] or [[login()]] to change the identity of the current user / login.
+     * 
+     * If [[enableAutoLogin]] is `true` and login cookie is present:
+     * - if [[autoRenewCookie]] is `true`, cookie is renewed with new [[authKey]] and original [[duration]]
+     * - if [[autoRenewCookie]] is `false`, cookie is destroyed because it's no longer valid
+     *
+     * @param IdentityInterface $identity the identity object associated with the currently logged user.
+     * @throws InvalidValueException if `$identity` object does not implement [[IdentityInterface]].
+     */
+    public function userAuthKeyUpdated($identity)
```

Comment:
There should be a better name. What it does:

1. Switch identity.
2. Refresh "remember me" identity cookie if auto login is enabled.

Reply:
Maybe these two should be separate?